mautrix/util/async_db/aiosqlite.py

```diff
@@ -125,12 +126,16 @@ def _add_missing_pragmas(init_commands: list[str]) -> list[str]:
                 has_foreign_keys = True
             elif "journal_mode" in cmd:
                 has_journal_mode = True
+            elif "synchronous" in cmd:
+                has_synchronous = True
             elif "busy_timeout" in cmd:
                 has_busy_timeout = True
         if not has_foreign_keys:
             init_commands.append("PRAGMA foreign_keys = ON")
         if not has_journal_mode:
             init_commands.append("PRAGMA journal_mode = WAL")
+        if not has_synchronous:
```

Comment:
Should probably check the exact journal mode for safety

```suggestion
        if not has_synchronous and "PRAGMA journal_mode = WAL" in init_commands:
```